File: app/src/main/java/com/samkeet/takeup/fragments/ImageFragment.java
package com.samkeet.takeup.fragments;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.samkeet.takeup.R;

/**
 * Created by Hi Abhishek on 5/9/2017.
 */

public class ImageFragment extends Fragment {
    private int mResourceId;

    public static ImageFragment newInstance(int resourceId){
        ImageFragment imageFragment = new ImageFragment();
        Bundle args = new Bundle();
        args.putInt("resourceId", resourceId);
        imageFragment.setArguments(args);
        return imageFragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mResourceId = getArguments().getInt("resourceId");
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragement_image_indicator, container, false);
        ImageView imageView = (ImageView) view.findViewById(R.id.image_view);
        imageView.setImageResource(mResourceId);
        return view;
    }
}

File: app/src/main/java/com/samkeet/takeup/fragments/ImageFragment.kt
package com.samkeet.takeup.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.samkeet.takeup.R

/**
 * Created by knightshade on 10/19/17.
 */
class ImageFragment : Fragment() {
    private var mResourceId: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mResourceId = arguments.getInt("resourceId")
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.fragement_image_indicator, container, false)
        val imageView = view.findViewById(R.id.image_view) as ImageView
        imageView.setImageResource(mResourceId)
        return view
    }

    companion object {

        fun newInstance(resourceId: Int): ImageFragment {
            val imageFragment = ImageFragment()
            val args = Bundle()
            args.putInt("resourceId", resourceId)
            imageFragment.arguments = args
            return imageFragment
        }
    }
}
